### ISSUE TITLE
Destructuring does not seem to work with test functions as they do with `tap`

### ISSUE THREAD

**Issue by dmitriz** [avajs/ava#1802]
I was expecting to treat the function inside test as ordinary function,
but the destructuring does not seem to work:

```js
const test = require('ava')

// this works
test('plain', t => {
	t.is(1+1, 2)
})

// this throws
test('plain', ({ is }) => {
	is(1+1, 2)
})
```

```sh
1 passed
  1 failed

  plain

  /Users/dmitrizaitsev/Dropbox/Sandbox/ava-test/destructuring_test.js:8

   7: test('plain', ({ is }) => {
   8:   is(1+1, 2)               
   9: })                         

  Error thrown in test:

  TypeError {
    message: 'Cannot read property \'_test\' of undefined',
  }
```

On the other hand, the same works fine with `tap` and `tape`:

```js
const { test } = require('tap')
// const { test } = require('tape')

test('plain', t => {
	t.equal(1+1, 2)
	t.end()
})

test('plain', ({ equal, end }) => {
	equal(1+1, 2)
	end()
})
```

Here is my env if it is of any use:
```sh
Node.js v8.9.4
darwin 12.6.0
npm 5.6.0
ava 0.25.0
```

**Comment by sindresorhus** [comment]
This was fixed in 1.0.0-beta.2: https://github.com/avajs/ava/releases/tag/v1.0.0-beta.2

**Comment by dmitriz** [comment]
Thanks, that is awesome!
